checkout/access-checkout-android/src/test/java/com/worldpay/access/checkout/validation/DateValidatorTest.kt

```diff
@@ -1,380 +1,348 @@
     }
 
     @Test
     fun `given partially valid single digit month and a completely valid year then should be partially  and completely invalid`() {
-        setCurrentDate(2019, 5 /* May */, 7)
+        val dateValidator = getValidatorWithDate(2019, 5, 7)
 
-        assertEquals(ValidationResult(partial = false, complete = false), validator.validate("1", "20"))
+        assertEquals(ValidationResult(partial = false, complete = false), dateValidator.validate("1", "20", CARD_CONFIG_NO_BRAND))
     }
 
     @Test
     fun `given invalid single digit month and a completely valid year then should be completely invalid`() {
-        setCurrentDate(2019, 5 /* May */, 7)
+        val dateValidator = getValidatorWithDate(2019, 5, 7)
 
-        assertEquals(ValidationResult(partial = false, complete = false), validator.validate("2", "20"))
+        assertEquals(ValidationResult(partial = false, complete = false), dateValidator.validate("2", "20", CARD_CONFIG_NO_BRAND))
     }
 
     @Test
     fun `given future date for next year then should be completely valid`() {
-        setCurrentDate(2019, 5 /* May */, 7)
+        val dateValidator = getValidatorWithDate(2019, 5, 7)
 
-        assertEquals(ValidationResult(partial = false, complete = true), validator.validate("11", "20"))
+        assertEquals(ValidationResult(partial = false, complete = true), dateValidator.validate("11", "20", CARD_CONFIG_NO_BRAND))
     }
 
     @Test
     fun `given future date in the current year then should be completely valid`() {
-        setCurrentDate(2019, 5 /* May */, 7)
+        val dateValidator = getValidatorWithDate(2019, 5, 7)
 
-        assertEquals(ValidationResult(partial = false, complete = true), validator.validate("11", "19"))
+        assertEquals(ValidationResult(partial = false, complete = true), dateValidator.validate("11", "19", CARD_CONFIG_NO_BRAND))
     }
 
     @Test
     fun `given past date in the current year then should be completely invalid`() {
-        setCurrentDate(2019, 5 /* May */, 7)
+        val dateValidator = getValidatorWithDate(2019, 5, 7)
 
-        assertEquals(ValidationResult(partial = false, complete = false), validator.validate("01", "19"))
+        assertEquals(ValidationResult(partial = false, complete = false), dateValidator.validate("01", "19", CARD_CONFIG_NO_BRAND))
     }
 
     @Test
     fun `given past date for last year then should be completely invalid`() {
-        setCurrentDate(2019, 5 /* May */, 7)
+        val dateValidator = getValidatorWithDate(2019, 5, 7)
 
-        assertEquals(ValidationResult(partial = false, complete = false), validator.validate("01", "18"))
+        assertEquals(ValidationResult(partial = false, complete = false), dateValidator.validate("01", "18", CARD_CONFIG_NO_BRAND))
     }
 
     @Test
     fun `given last day of a month then should be completely valid`() {
```

Comment:
Out of curiosity, do you know why we have this test? It looks really weird to be testing the outcome of the validation depending on today's date

Reply:
no idea unfortunately